packages/core/src/utils/SimpleStore.ts

```diff
@@ -0,0 +1,37 @@
+import * as React from "react";
+import { SimpleEventEmitter } from "./SimpleEventEmitter";
+
+/**
+ * A simple store that can be updated anywhere, with hook-support.
+ *  - Used to hold state (like colorScheme preference), which can be updated from a single
+ *    event listener, and have those updates emitted out to multiple hook-usages.
+ */
+export class SimpleStore<S> {
```

Comment:
THE NEXT REDUX!

Jk, this is like the most barebones possible state-management solution. I like it because it's lean, and doesn't require `Context`.